File: Java/src/main/java/Exceptions/CircularDependencyException.java
package Exceptions;

import java.lang.reflect.Type;

/**
 * Created by user on 22/9/2015.
 */
public class CircularDependencyException extends RuntimeException {
    public CircularDependencyException(Type contract) {
        super("Circular dependency detected when trying to resolve: " + contract.getTypeName());
    }
}

File: Container/Source/Exceptions/CircularDependencyException.cs
using System;

namespace IoC.Source.Exceptions
{
    /// <summary>
    /// Indicates that a type is trying to be resolved that has a circular dependency in the dependency chain.
    /// </summary>
    internal class CircularDependencyException : Exception
    {
        public CircularDependencyException(Type contract) 
            : base($"Circular dependency detected when trying to resolve: {contract.FullName}")
        {
        }
    }
}
